### PR TITLE
SRS document aka homework week 03

### PR DESCRIPTION
**Pull request by hottek**
## First draft of SRS document
Please review the document, provide feedback, fix spelling or parts you don't like / don't agree with. If necessary open an issue on YouTrack (e.g. in column to be discussed).
### Todo
- review
- correct linking of UseCase png

### CODE REVIEW

File: doc/SRS.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+#### 3.5.1 Coding standards
+In order to maintain supportability and readability of our code, we will adopt the clean code standard, common naming conventions uniform formatting and best practices throughout the project.
+#### 3.5.2 Maintenance Utilities
+In order to test and integrate newer versions of the application, a continuous integration service is required.
+
+### 3.6 Design Constraints
+Our goal is to provide a modern design in both code and application.  
+Server-side programs ought to be compiled using Golang. A REST-API will be used to communicate between client and server.  
+The dashboard will be implemented using a MVC architecture.  
+The chrome-plugin and the android autofill service will be implemented using their respective technologies and languages.
```
Comment (zkdev) [inline doc/SRS.md]: - chrome plugin

File: doc/SRS.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+- Frontend development: tbd
+- Project planning tool: JetBrains YouTrack
+- Build management: tbd
+- CI: tbd
+
+### 3.7 On-line User Documentation and Help System Requirements
+As stated our goal is to make our application as intuitive as possible. However, we will provide a FAQ and documentation. This will be especially helpful for users who want to know how our application works in depth.
+
+### 3.8 Purchased Components
+A server hosted at [zap-hosting.com](https://zap-hosting.com/de/) is used to run the frontend application as well as the backend server.
```
Comment (zkdev) [inline doc/SRS.md]: maybe restructure this section
- domain at alldomains.hosting
- two servers hosted at zap
- we mentioned a KeyCloud@zeekay.dev mail already, mail server hosted at netcup

#forthesakeofdetails

File: doc/SRS.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+### 3.7 On-line User Documentation and Help System Requirements
+As stated our goal is to make our application as intuitive as possible. However, we will provide a FAQ and documentation. This will be especially helpful for users who want to know how our application works in depth.
+
+### 3.8 Purchased Components
+A server hosted at [zap-hosting.com](https://zap-hosting.com/de/) is used to run the frontend application as well as the backend server.  
+Currently there are no other purchased components.
+
+### 3.9 Interfaces
+#### 3.9.1 User Interfaces
+There will be following interfaces available:
```
Comment (zkdev) [inline doc/SRS.md]: - "Following interfaces will be available"

File: doc/SRS.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+- Dashboard: manage (add, remove), retrieve passwords, change settings
+- Chrome-Plugin: manage, retrieve passwords
```
Comment (zkdev) [inline doc/SRS.md]: chrome plugin

File: doc/SRS.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+#### 3.9.2 Hardware Interfaces
+n/a
+#### 3.9.3 Software Interfaces
+As the dashboard is a web-application it should be running on the most common browsers. Nevertheless, we want to primarily support the Google Chrome browser and its features.
+#### 3.9.4 Communications Interfaces
+The client will connect to the server `https`. An unencrypted connection wil not be supported.
```
Comment (zkdev) [inline doc/SRS.md]: - "The client-server communication uses the encrypted `HTTPS` protocol. Unencrypted connections are not supported"

File: doc/SRS.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+#### 3.9.3 Software Interfaces
+As the dashboard is a web-application it should be running on the most common browsers. Nevertheless, we want to primarily support the Google Chrome browser and its features.
+#### 3.9.4 Communications Interfaces
+The client will connect to the server `https`. An unencrypted connection wil not be supported. 
+
+### 3.10 Licensing Requirements
+Under [MIT license](https://github.com/zkdev/keycloud/blob/master/LICENSE.md).
+
+### 3.11 Legal, Copyright, and Other Notices
+The KeyCloud team will not take any responsibility for the loss of personal data. The KeyCloud logo may only be used for the official KeyCLoud applications.
```
Comment (zkdev) [inline doc/SRS.md]: - "KeyCLoud"

**Comment by zkdev** [comment]
we are hopping between
- "we are using"
and
- "we will use"

not sure

File: doc/SRS.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+Software Requirements Specification | a document, which captures the complete software requirements for the system, or a portion of the system
+### 1.4 References
+Title | Date | Publishing organization |  
+--- | :---:  | ---
+[KeyCloud Blog](https://keycloud.zeekay.dev/) | 12.10.2019 | KeyCloudTeam  
+[YouTrack Instance](https://keycloud-dev.zeekay.dev:7000/issues) | 12.10.2019 | KeyCloudTeam  
+[SRS](../doc/SRS.md) | 12.10.2019 | KeyCloudTeam  
+
+### 1.5 Overview
+The remainder is structured in the following way: In the next chapter, the overall description, an overview of the functionality and an use-case-diagram is given.
```
[the remark this answers is not in the document]
Comment (hottek) [reply doc/SRS.md]: checks out

File: doc/SRS.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+### 3.1 Functionality
+This section will list all functional requirements for KeyCloud and explain their functionality.
+Each of the following subsections represents a subsystem of our application.
+#### 3.1.1 Dashboard
+By using the dashboard several activities can be managed.  
+First of all there will be the option to create a user account. Furthermore the passwords of a user are managed here. Settings can also be changed.  
+Additional features might include the generation of passwords.
+#### 3.1.2 Chrome-Plugin
+The chrome-plugin will enable the user to get his passwords in the Google Chrome browser.
+Additional features might include the function to add and remove individual passwords.
```
[the remark this answers is not in the document]
Comment (hottek) [reply doc/SRS.md]: no

**Pull request merged**